lib/goth.ex

```diff
@@ -33,9 +36,11 @@ defmodule Goth do
       that it is tried to be automatically refreshed. Defaults to
       `#{@refresh_before_minutes * 60}` (#{@refresh_before_minutes} minutes).
 
-    * `:http_client` - a `{module, opts}` tuple, where `module` implements the
-      `Goth.HTTPClient` behaviour and `opts` is a keywords list to initialize the client with.
-      Defaults to `{Goth.HTTPClient.Hackney, []}`.
+    * `:http_client` - a `{&fun/1, opts}` tuple or `&fun/1`, where `fun` is a
+      function reference with arity 1 and `opts` is a keywords list to initialize the client with.
+      The function will receive a keyword list with required fields: `:method`, `:url`, `:body` and
+      `:headers`, and `opts` merged into it, if available. And the function should return
+      `{:ok, map} | {:error, exception}`. Defaults to `{&Goth.__hackney__/1, []}`.
```

Comment:
`Goth.__hackney__` is hidden so we shouldn't reference it from other docs. I think we could have a special value, atom `:hackney`, that means the built-adapter.

Here's a stab at this section:

```markdown
    * `:http_client` - a function that makes the HTTP request. Can be one of the following:

        * `fun` - same as `{fun, []}`

        * `{fun, opts}` - `fun` must be a 1-arity function that receives a keyword list with fields
          `:method`, `:url`, `:headers`, and `:body` along with any passed `opts`. The function must return
          `{:ok, %{status: status, headers: headers, body: body}}` or `{:error, exception}`.
          Example: `{&HTTPClient.request/1, connect_timeout: 5000}`.

          `fun` can also be an atom `:hackney` to use the built-in [Hackney](http://github.com/benoitc/hackney)-based
          client.

      Defaults to `{:hackney, []}`
```

wdyt?

Reply:
I like this idea, I'll make that change

Reply:
Also, added a case when you only use `:hackney` without options